### PR TITLE
Add database to bot

### PR DESCRIPTION
**Pull request by blaumeise20**
Finally we have a database in our bot! Now we can use it in the modules to store data and create cool commands.

### CODE REVIEW

File: DiscordBot.Domain/Database/DatabaseHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            byte[] primaryKeyHash;
+            if (props.PrimaryKeys.Count > 0)
+            {
+                StringBuilder builder = new StringBuilder();
+                foreach (var key in props.PrimaryKeys)
+                {
+                    builder.Append(key.GetMethod.Invoke(storeObject, null)?.ToString() ?? "null");
+                }
+
+                primaryKeyHash = MD5.Create().ComputeHash(Encoding.UTF8.GetBytes(builder.ToString()));
```
Comment (JKamsker) [inline DiscordBot.Domain/Database/DatabaseHelpers.cs:42]: Please dispose the CSP

File: DiscordBot.Domain/Database/DatabaseHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                storeObject.PartKey = Convert.ToBase64String(primaryKeyHash, 0, 5);
+                storeObject.Id = Convert.ToBase64String(primaryKeyHash, 5, primaryKeyHash.Length - 5);
+            }
+        }
+
+        private static readonly Dictionary<Type, ObjectStoreProperties> _knownTypes = new Dictionary<Type, ObjectStoreProperties>();
+        public static ObjectStoreProperties GetProperties<T>() where T : class
+        {
+            if (!_knownTypes.ContainsKey(typeof(T)))
```
Comment (JKamsker) [inline DiscordBot.Domain/Database/DatabaseHelpers.cs:63]: Use TryGetValue

File: DiscordBot.Domain/Database/DatabaseContainer.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using DiscordBot.Domain.Configuration;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Options;
+
+namespace DiscordBot.Domain.Database
```
Comment (JKamsker) [inline DiscordBot.Domain/Database/DatabaseContainer.cs:7]: This is in a complete wrong namespace

File: DiscordBot.Modules/CommandModules/CodeSnippetModule.cs [hunk 4]
```diff
@@ -1,37 +1,36 @@
...
+
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
-using Newtonsoft.Json;
 
 namespace DiscordBot.Modules.CommandModules
 {
     [Group("code")]
-    [Summary("Saves and shows code snippets.")]
+    [Summary("Speichert und zeigt Code-Snippets an.")]
```
Comment (JKamsker) [inline DiscordBot.Modules/CommandModules/CodeSnippetModule.cs:17]: Nono

File: DiscordBot.Domain/Database/DatabaseContainer.cs (hunk 3 above)
Comment (blaumeise20) [reply DiscordBot.Domain/Database/DatabaseContainer.cs:7]: Should I use DiscordBot.Domain.Abstractions?

Comment (JKamsker) [reply DiscordBot.Domain/Database/DatabaseContainer.cs:7]: Code doesnt belong into the Domain namespace. It all should be moved to something like DiscordBot.DataAccessLayer or DiscordBot.Dal

File: DiscordBot.Modules/CommandModules/CodeSnippetModule.cs (hunk 4 above)
Comment (blaumeise20) [reply DiscordBot.Modules/CommandModules/CodeSnippetModule.cs:17]: But this is a German bot 😀 So, not really German, but its language is German. For example take a look at GrillBot. It is also a Discord bot written in C#. Do you understand anything: https://github.com/Misha12/GrillBot/blob/master/Grillbot/Modules/AdminModule.cs#L32 ?

**Comment by blaumeise20** [comment]
Something went wrong with this PR, I will close it and open a new one 😀

**Pull request closed without merging**